src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Waiting/VisualStudioWaitContext.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Runtime.InteropServices;
+using System.Threading;
+
+using Microsoft.VisualStudio.ProjectSystem.Waiting;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Waiting
+{
+    internal partial class VisualStudioWaitContext : IWaitContext
```

Comment:
Is this another copy of the Roslyn version of this?

Reply:
Slightly different but essentially the same. I am not aware of any reasonable public API to do this.

Reply:
The editor *also* made a copy of this; but exposed an IUIThreadOperation API. Not sure if that's directly usable but would be good to sort out.